### ISSUE TITLE
Is it possible to get RequestBody when RetrofitException?

### ISSUE THREAD

**Issue by tonyjs** [square/retrofit#1171]
Lately I got the many many 400 bad request error posting, so I want to know the real body when catch the exception.

Is there a way to get the RequestBody when catch the RetrofitException?

**Comment by JakeWharton** [comment]
Do you mean `RetrofitError`? There is no class called `RetrofitException`.

You can read the response body by calling `getBodyAs(SomeType.class)` to have it converted or if you want the raw body you can use `getResponse().body()`.

**Comment by tonyjs** [comment]
Thanks for your answer. Yeah right that is RetrofirError :)

I saw the log what is the request body When request time.
for example :

{ "token" : "123-456-789" }

That is what I want to send to Server.
But when I see the Server log, I can find strange value of body.
for example :

{ "token" : "123-456-789/" }

So Server throws the 400 error.
Then I asked to you that Can I find the request body when the Exception has
thrown.

so... I got a question.

ah! I use the Converter what using Jackson, and the token(for request body)
is from the SharedPreferences.

Is it possble to modulation when write body into possible?
or Is it possible to get strange value from SharedPreferences with multi
thread Not multi process?

I'm so Sorry that let me spend your time.
2015. 10. 8. 오전 1:06에 "Jake Wharton" <notifications@github.com>님이 작성:
